floodlight/io/statsperform.py

```diff
@@ -399,55 +482,51 @@ def read_open_tracking_data_csv(
         # update codes
         codes["possession"][segment] = ball_df["possession"].values[appearance]
 
-    # create XY objects
-    home_ht1 = XY(xy=xydata["Home"][0], framerate=10)
-    home_ht2 = XY(xy=xydata["Home"][1], framerate=10)
-    away_ht1 = XY(xy=xydata["Away"][0], framerate=10)
-    away_ht2 = XY(xy=xydata["Away"][1], framerate=10)
-    ball_ht1 = XY(xy=xydata["Ball"][0], framerate=10)
-    ball_ht2 = XY(xy=xydata["Ball"][1], framerate=10)
-
-    # create Code objects
-    poss_ht1 = Code(
-        name="possession",
-        code=codes["possession"][0],
-        definitions=dict([(team_id, team) for team, team_id in team_ids.items()]),
-        framerate=10,
-    )
-    poss_ht2 = Code(
-        name="possession",
-        code=codes["possession"][1],
-        definitions=dict([(team_id, team) for team, team_id in team_ids.items()]),
-        framerate=10,
-    )
+    # create objects
+    xy_objects = {}
+    possession_objects = {}
+    for segment in segments:
```

Comment:
possession_objects should be Dict[str, Dict[str, XY]] but segments are given as integer. Either change the docstring or convert segment to str.